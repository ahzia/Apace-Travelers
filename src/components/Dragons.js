import { useState, useEffect } from 'react';
import { useDispatch } from 'react-redux';
import { fetchPostsRequestDragons } from '../redux/dragons/dragons';
import store from '../redux/configureStore';

const dragons = () => {
  const [dragons, setdragons] = useState([]);
  const dispatch = useDispatch();

  useEffect(() => {
    store.subscribe(() => {
      setdragons(store.getState().dragonReducer.dragons);
    });
  });

  useEffect(() => {
    dispatch(fetchPostsRequestDragons());
  }, []);

  return (
    <div className="container d-flex flex-column mx-5 my-2">
      {dragons.map((dragon) => (
        <div key={dragon.dragon_id} className="row d-flex my-2">
          <img
            className="col-md-3"
            src={dragon.flickr_images[0]}
            alt={dragon.dragon_name}
          />
          <div className="col-md-9">
            <h5>{dragon.dragon_name}</h5>
            <p>{dragon.description}</p>
            <button type="button" className="btn btn-primary">
              Reserve dragon
            </button>
          </div>
        </div>
      ))}
    </div>
  );
};

export default dragons;
