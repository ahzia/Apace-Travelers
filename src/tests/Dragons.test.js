import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import Dragons from '../components/Dragons';
import store from '../redux/configureStore';

describe('Component test snapshot', () => {
  it('Dragons renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Dragons />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
