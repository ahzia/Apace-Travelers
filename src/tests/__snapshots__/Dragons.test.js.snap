// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component test snapshot Dragons renders correctly 1`] = `
<div
  className="container d-flex flex-column mx-5 my-2"
/>
`;
